rfcs/45-stageless.md

```diff
@@ -0,0 +1,564 @@
+Each frame, the `App`'s `runner` function will run the schedule,
+causing the **scheduler** to run the systems in parallel using a strategy that respects all of the configured constraints (or panic, if it is **unsatisfiable**).
+
+In the beginning, each `Schedule` is entirely unordered: systems will be selected in an arbitrary order and run if and only if all of the data that it must access is free.
+Just like with standard borrow checking, multiple systems can read from the same data at once, but writing to the data requires an exclusive lock.
+Systems which cannot be run in parallel are said to be **incompatible**.
+
+Bevy controls the ordering of its schedule(s) The solution via **intent-based configuration**: record the exact, minimal set of constraints needed for a scheduling strategy to be valid, and then let the scheduler figure it out!
+As the number of systems in your app grows, the number of possible scheduling strategies grows exponentially, and the set of *valid* scheduling strategies grows almost as quickly.
+However, each system will continue to work hand-in-hand with a small number of closely-related systems, interlocking in a straightforward and comprehensible way.
+
+By configuring our systems with straightforward, local rules, we can allow the scheduler to pick one of the possible valid paths, optimizing as it sees fit without breaking our logic.
+Just as importantly, we are not over-constraining our ordering, allowing subtle ordering bugs to surface quickly and then be stamped out with a well-considered rule.
+
+### The `App` stores multiple schedules
+
+The `App` can store multiple `Schedules` in a `HashMap<Box<dyn ScheduleLabel>, Schedule>` storage.
+This is used for the enter and exit schedules of states, but can also be used to store, mutate and access additional schedules.
+You can even mutate the schedule that you are in, although you cannot mutate systems that are currently running (failing with an error), as that would be UB.
+
+The main and startup schedules can be accessed using the `CoreSchedule::Main` and `CoreSchedule::Startup` labels respectively.
+By default systems are added to the main schedule.
+You can control this by adding the `.to_schedule(MySchedule::Variant)` system descriptor to your system.
+
+You can access the schedules stored in the app using the `&Schedules` or `&mut Schedules` system parameters.
+Unsurprisingly, these never conflict with entities or resources in the `World`, as they are stored one level higher.
+
+#### Startup systems
+
+Startup systems are stored in their own schedule, with the `CoreSchedule::Startup` label.
+When using the runner added by `MinimalPlugins` and `DefaultPlugins`, this schedule will run exactly once on app startup.
+
+You can add startup systems with the `.add_startup_system(my_system)` method on `App`, which is simply sugar for `.add_system(my_system.to_schedule(CoreSchedule::Startup))`.

```

Comment:
I dislike how easily we can fall into the trap of https://github.com/bevyengine/bevy/discussions/2804 for documentation, even if in a real project we'd avoid this; I suppose this is because in real projects, systems are verbs, whereas in documentation you want to point at them, and hence treat them as nouns.